lumen/src/main.rs

```diff
@@ -25,16 +26,26 @@ fn setup_logger() {
 }
 
 
-async fn handle_client<S: AsyncRead + AsyncWrite + Unpin>(db: &db::Database, mut stream: S) -> Result<(), rpc::Error> {
+async fn handle_client<S: AsyncRead + AsyncWrite + Unpin>(db: &db::Database, mut stream: S, config: &Config) -> Result<(), rpc::Error> {
```

Comment:
Both `config` and `db` are contained in `SharedState_` that is created in `main`, it would be better to pass the one object that contains both...